Flagsmith.FlagsmithClient/FlagsmithConfiguration.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+namespace Flagsmith
+{
+    public class FlagsmithConfiguration
+    {
+        public FlagsmithConfiguration()
+        {
+            ApiUrl = "https://api.flagsmith.com/api/v1/";
```

Comment:
Does this set the default for all constructors or only if you use the 'no args constructor' that is defined here? Can we not set the default on the attribute definition below?

Reply:
we only have default constructor ,which just initialized default values of properties.we can set on attribute level but conventionally dotnet recommend it to be initialized at constructor level.

Reply:
But, ok, what happens if the user does this: 

```dotnet
var configuration = FlagsmithConfiguration(EnvironmentKey="my-key")
```

Will it pick up the default API url?

Reply:
No, as its no parameter constructor , user won't be allowed to pass his own params inside.

Reply:
Then this solution really doesn't work. Most of our users will not want to care about specifying the ApiUrl but they obviously will need to specify the environment key. How can we make it such that this is possible?

Reply:
I am removing `FlagsmithConfiguration.cs` and taking params directly from `flagsmithClient` constructor as we did in python implementation

Reply:
Ah, right, sorry! I forgot about the properties json approach in .NET. If that is a standard way to accomplish the initialisation here then I'm happy to stick with that approach.